### PR TITLE
fix: astro:env secret in middleware

### LINKED ISSUE #12324: astro:env can't find secrets in middleware
### Astro Info

```block
N/A
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

When using a secret inside a middleware, it now fails

1. `npm run build`
2. `npm run preview`
3. visit root

### What's the expected result?

Should still work

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-8b8hjn

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by florian-lefebvre**
## Todo

- [ ] remove error in base pipeline. Probably allows removing env info in ssr manifest

## Changes

- Closes #12324

## Testing

Added a middleware to test, tests should still pass. I also tested in a private affected project

## Docs

N/A

### CODE REVIEW

File: packages/astro/src/core/build/plugins/plugin-ssr.ts [hunk 1]
```diff
@@ -283,7 +283,7 @@ function generateSSRCode(settings: AstroSettings, adapter: AstroAdapter, middlew
 		`import { renderers } from '${RENDERERS_MODULE_ID}';`,
 		`import * as serverEntrypointModule from '${ADAPTER_VIRTUAL_MODULE_ID}';`,
 		`import { manifest as defaultManifest } from '${SSR_MANIFEST_VIRTUAL_MODULE_ID}';`,
-		edgeMiddleware ? `` : `import { onRequest as middleware } from '${middlewareId}';`,
+		// edgeMiddleware ? `` : `import { onRequest as middleware } from '${middlewareId}';`,
```
Comment (florian-lefebvre) [inline packages/astro/src/core/build/plugins/plugin-ssr.ts:286]: This solves the error of not getting preview working at all, but now causes an error 500

Comment (florian-lefebvre) [reply packages/astro/src/core/build/plugins/plugin-ssr.ts:286]:
```suggestion
```

**Comment by hckhanh** [comment]
I get into this bug too. I am still waiting for the patch.

**Comment by florian-lefebvre** [comment]
It's tricky so if how much time it will require ro be fixed. In the meantime you can downgrade to 4.16.5 I think

**Comment by hckhanh** [comment]
> It's tricky so idk how much time it will require to be fixed. In the meantime you can downgrade to 4.16.5 I think

I tried 4.16.5 but I still get this error

**Comment by ematipico** [comment]
@hckhanh please refrain from adding comments to the PR, unless they are concerns regarding the **code and implementation of the fix**. Feel free to follow up in the relative issue.

**Comment by florian-lefebvre** [comment]
!snapshot env-middleware

**Comment by florian-lefebvre** [comment]
!snapshot env-middleware

**Pull request closed without merging**